fast_llm/engine/training/config.py

```diff
@@ -90,71 +141,109 @@ class WandbConfig(Config):
 
 
 @config_class()
-class ValidationConfig(get_interval_config_class("validation")):
+class ValidationConfig(IntervalConfig):
+    interval = FieldUpdate(
+        desc="The number of training iterations between each validation phase."
+        " Setting to None will disable validation."
+    )
+    offset = FieldUpdate(desc="Offset for the first validation phase.")
     iterations: int | None = Field(
         default=None,
         desc="Number of iterations for each validation phase. Setting to None will disable.",
         hint=FieldHint.feature,
         valid=skip_valid_if_none(check_field(Assert.gt, 0)),
     )
 
-    def get_completed_iterations(self, training_iterations: int, completed_validations: int = 0):
+    def get_iteration_count(self, training_iterations: int, extra_validations: int = 0):
         # Number of completed validation iterations
-        return (
-            (training_iterations // self.interval + completed_validations) * self.iterations if self.enabled() else 0
-        )
+        return (self.get_count(training_iterations) + extra_validations) * self.iterations if self.enabled() else 0
 
 
 @config_class()
-class CheckpointConfig(get_interval_config_class("checkpoint")):
+class CheckpointBaseConfig(IntervalConfig):
+    _abstract = True
+    save_name: typing.ClassVar[str] = "save"
+    directory_name: typing.ClassVar[str] = "save"
```

Comment:
for what kind of checkpoint will we have `save` as the name of the output directory?

Reply:
None, that's a placeholder (I could just remove?)